spec/jobs/socure_shadow_mode_proofing_job_spec.rb

```diff
@@ -0,0 +1,384 @@
+              transaction_id: 'state-id-mock-transaction-id-456',
+              vendor_name: 'StateIdMock',
+              verified_attributes: [],
+            },
+            threatmetrix: {
+              client: nil,
+              success: true,
+              errors: {},
+              exception: nil,
+              timed_out: false,
+              transaction_id: 'ddp-mock-transaction-id-123',
+              review_status: 'pass',
+              response_body: {
+                "fraudpoint.score": '500',
+                request_id: '1234',
+                request_result: 'success',
+                review_status: 'pass',
+                risk_rating: 'trusted',
+                summary_risk_score: '-6',
+                tmx_risk_rating: 'neutral',
+                tmx_summary_reason_code: ['Identity_Negative_History'],
+                first_name: '[redacted]',
+              },
+            },
+          },
+        },
+        ssn_is_unique: true,
+      },
+    )
+  end
+
+  let(:socure_idplus_base_url) { 'https://example.org' }
+
+  before do
+    document_capture_session.store_proofing_result(proofing_result.to_h)
+
+    allow(IdentityConfig.store).to receive(:socure_idplus_base_url).
+      and_return(socure_idplus_base_url)
+  end
+
+  describe '#perform' do
+    subject(:perform) do
+      allow(job).to receive(:create_analytics).and_return(analytics)
+
+      job.perform(
+        document_capture_session_result_id:,
+        encrypted_arguments:,
+        service_provider_issuer:,
+        user_email:,
+        user_uuid:,
+      )
+    end
+
+    let(:analytics) do
+      FakeAnalytics.new
+    end
+
+    let(:socure_response_body) do
+      {
+        referenceId: 'a1234b56-e789-0123-4fga-56b7c890d123',
+        kyc: {
+          reasonCodes: [
+            'I919',
+            'I914',
+            'I905',
+          ],
+          fieldValidations: {
+            firstName: 0.99,
+            surName: 0.99,
+            streetAddress: 0.99,
+            city: 0.99,
+            state: 0.99,
+            zip: 0.99,
+            mobileNumber: 0.99,
+            dob: 0.99,
+            ssn: 0.99,
+          },
+        },
+        customerProfile: {
+          customerUserId: '129',
+          userId: 'u8JpWn4QsF3R7tA2',
+        },
+      }
+    end
+
+    before do
+      stub_request(:post, 'https://example.org/api/3.0/EmailAuthScore').
+        to_return(
+          headers: {
+            'Content-Type' => 'application/json',
+          },
+          body: JSON.generate(socure_response_body),
+        )
+    end
+
+    context 'when document_capture_session_result_id is valid' do
+      it 'makes a proofing call' do
+        expect(job.proofer).to receive(:proof).and_call_original
+        perform
+      end
+
+      it 'does not log an idv_socure_shadow_mode_proofing_result_missing event' do
+        perform
+        expect(analytics).not_to have_logged_event(:idv_socure_shadow_mode_proofing_result_missing)
```

Comment:
We did not previously support `not_to have_logged_event` -- that's what the update to `HaveLoggedEventMatcher` (below) is about.

Reply:
Thanks for adding it!